src/Illuminate/Routing/Concerns/HasCustomAttributes.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Illuminate\Routing\Concerns;
+
+use Illuminate\Support\Collection;
+use ReflectionAttribute;
+use ReflectionMethod;
+
+trait HasCustomAttributes
```

Comment:
Attributes can be pretty arcane for the average Laravel dev, a brief introduction can be very helpful:
```suggestion
/**
 * Trait HasCustomAttributes
 *
 * This trait allows Laravel controllers to handle custom PHP attributes on controller methods.
 * Attributes are metadata annotations that can be added to classes, methods, or properties.
 * They provide additional context or functionality and can be retrieved at runtime using reflection.
 *
 * This trait specifically retrieves attributes from controller methods and invokes their `handle` method (if it exists)
 * before executing the controller action. Attributes must implement the `ICustomAttribute` interface to be processed.
 *
 * For more information on PHP attributes, see the official documentation:
 * @link https://www.php.net/manual/en/language.attributes.overview.php
 *
 * Example usage:
 * ```php
 * #[SomeCustomAttribute]
 * public function myControllerMethod()
 * {
 *     // Action logic here
 * }
 * ```
 */
trait HasCustomAttributes
```

Reply:
Such huge comments are a little unusual for Laravel—might be better suited for the docs 🤔